src/vtk_calculator.py

```diff
@@ -18,55 +18,101 @@
 
 Scalar calculation and writing to given file
 
-./vtk_calculator.py inputmesh.vtk exp(cos(x)+sin(y)) -t e^(cos(x)+sin(y)) -o outputmesh.vtk
+vtk_calculator.py -m inputmesh.vtk -f "exp(cos(x)+sin(y))" --data "e^(cos(x)+sin(y))" -o outputmesh.vtk
 
 Vector field and appends to input mesh
 
-./vtk_calculator.py inputmesh.vtk x*iHat+cos(y)*jHat-sin(z)*kHat -t MyVectorField
+vtk_calculator.py -m "inputmesh.vtk" -f "x*iHat+cos(y)*jHat-sin(z)*kHat" -d "MyVectorField"
 
 There is also a diff mode which provides statistic between input data and function calculated
 (Note that it only works for scalar data)
 
-./vtk_calculator.py -m inputmesh.vtu -f x+y -d mydata --diff --stats
+For example to calculate difference between given function "x+y"
+and data "mydata" and override the result to "mydata" and to save statistics into a file
+following command is used.
 
-Calculates difference between given function and mydata data save over rides into variable data and saves statistics
+vtk_calculator.py -m inputmesh.vtu -f "x+y" -d "mydata" --diff --stats
 
-./vtk_calculator.py -m inputmesh.vtu -f x+y -d diffence -diffd mydata --diff
-
-Calculates difference between given function and mydata data save into diffence data
+If don't want to override "mydata" instead save into another variable "difference" following command is used.
 
+vtk_calculator.py -m inputmesh.vtu -f "x+y" -d "difference" --diffdata "mydata" --diff --stats
 """
 
 
 def parse_args():
     parser = argparse.ArgumentParser(description=__doc__)
-    parser.add_argument("--mesh", "-m", dest="in_meshname", required=True,
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("--mesh", "-m", dest="in_meshname",
                         help="The mesh (VTK Unstructured Grid) used as input")
-    parser.add_argument("--function", "-f", dest="function", required=True,
+    parser.add_argument("--function", "-f", dest="function", default="eggholder3d",
                         help="""The function to evalutate on the mesh.
-            Syntax is the same as used in the calculator object, coordinates are given as e.g.  'cos(x)+y'.""")
+            Syntax is the same as used in the calculator object, coordinates are given as e.g.  'cos(x)+y'.
+            Default is Egg-Holder function in 3D.
+            Alternatively, you can use predefined function """)
```

Comment:
We should also mention the `--functions` option here and put the `--functions` option immediately below the `--function` option.

Reply:
Would it be possible to print the list of available functions in case someone misspells a function name on the command line, i.e., catch the error and say, available options are ...

Reply:
Ok then let's skip this for the moment.